WcaOnRails/app/models/competition.rb

```diff
@@ -376,6 +380,10 @@ def warnings_for(user)
         warnings[:announcement] = I18n.t('competitions.messages.not_announced')
       end
 
+      if (self.registration_open - self.announced_at) > 172_800
```

Comment:
If the competition is not announced you subtract `nil` here. I think it would be more clear move this outside the main `if` into an outer one like this:

```rb
if self.announced?
  # one check
else
  # the other check
end
```

Reply:
I don't think this issue would error come up. showAtAll is when WCAT has approved the competition and made it visible. The announcement button is pressed a few seconds after.

Note to self: There needs to also be a check for if the results have been posted. There a lot of past competitions which had registration open in less than the currently required 48h after announcement and they don't need to see this error.

I don't have an opinion on your concrete proposal, I just built on top of what was already in the code.

Reply:
Between approving a competition and pressing he announcement button there is a refresh, in which case I can imagine the page crashing ^^ Also now I see some tests fail because `registration_open` may be `nil`, so it would be best to check for that. As this is itself a bit of logic, I'd still suggest moving it outside the main `if`.

```rb
# the first condition ensures the registration range fields are other than nil,
# the second condition definitely helps to ignore past competitions, but makes more sense

if registration_range_specified? && !registration_past?
  if self.announced?
    # one check
  else
    # the other check
  end
end

# somewhere below

def registration_range_specified?
  registration_open.present? && registration_close.present?
end
```

Reply:
I tried implementing what you sent now! Sorry for the delay, I hope it works this time. :)